core/market/decentralized/src/db/models/proposal.rs

```diff
@@ -141,6 +122,56 @@ impl Proposal {
         }
     }
 
+    pub fn new_provider_initial(demand: ModelDemand, offer: ModelOffer) -> Proposal {
+        let negotiation = Negotiation::new(&demand, &offer, OwnerType::Provider);
+        let proposal_id = ProposalId::generate_id(
+            &offer.id,
+            &demand.id,
+            &demand.creation_ts,
+            OwnerType::Provider,
+        );
+
+        let proposal = DbProposal {
+            id: proposal_id,
+            prev_proposal_id: None,
+            negotiation_id: negotiation.id.clone(),
+            properties: demand.properties,
+            constraints: demand.constraints,
+            state: ProposalState::Draft,
+            creation_ts: demand.creation_ts,
+            expiration_ts: demand.expiration_ts,
+        };
+
+        Proposal {
+            body: proposal,
+            negotiation,
+        }
+    }
+
+    pub fn counter_with(mut self, proposal: &ClientProposal) -> Proposal {
```

Comment:
This status will be always None, since it was created by agent

Reply:
I've meant the status of ClientProposal which we've got from REST API, so probably it can and should be checked, but this is minor

Reply:
Yes I meant the same status ;)
We ignore this status. It will be None probably. We set our internal status always to Draft.
Do you want to check if agent didn't set it to something?